src/julia-parser.scm

```diff
@@ -1661,7 +1661,10 @@
   (with-space-sensitive
    (if (eq? (peek-token s) '|.|)
        (begin (take-token s) '__dot__)
-       (parse-atom s #f))))
+       (let ((name (parse-atom s #f)))
+         (if (and (pair? name) (eq? (car name) 'call))
```

Comment:
I think this can be much stronger and allow only a symbol. Also the error message "invalid macro usage" in `macroify-name` should probably be changed to match this one; that would be clearer since it seems to me that error can only apply to the name anyway.

Reply:
Most caller's next call `(or (symbol? ex) (valid-modref? ex)` on this result, but one case firsts calls `parse-call-chain`, then calls `macroify-call` on the result instead.

Reply:
Sure I'm happy to change that. Note that this would also disallow the following weird-but-working syntax:

```
julia> Meta.parse("@(Mod.mac) x y")
:(#= none:1 =# Mod.@mac x y)
```

How about disallowing parentheses after `@` entirely to avoid pointless forms like

```
julia> Meta.parse("@(mac) x y")
:(#= none:1 =# @mac x y)
```

and abominations such as

```
julia> Meta.parse("@(((((mac))))) x y")
:(#= none:1 =# @mac x y)
```

Making the error messages more consistent is good too.